Setting LV_DRAW_SW_COMPLEX 0 won't compile

### LVGL version

v9.0.0

### What happened?

If I try to compile with LV_DRAW_SW_COMPLEX 0, build errors in lv_draw_sw_box_shadow.c will occur

[ 24%] Building C object lvgl/CMakeFiles/lvgl.dir/src/draw/sw/lv_draw_sw_box_shadow.c.o
/home/dha/Software/lvgl/lvgl/src/draw/sw/lv_draw_sw_box_shadow.c: In function ‘lv_draw_sw_box_shadow’:
/home/dha/Software/lvgl/lvgl/src/draw/sw/lv_draw_sw_box_shadow.c:120:5: warning: implicit declaration of function ‘shadow_draw_corner_buf’ [-Wimplicit-function-declaration]
  120 |     shadow_draw_corner_buf(&core_area, (uint16_t *)sh_buf, dsc->width, r_sh);
      |     ^~~~~~~~~~~~~~~~~~~~~~
/home/dha/Software/lvgl/lvgl/src/draw/sw/lv_draw_sw_box_shadow.c:128:5: error: unknown type name ‘lv_draw_sw_mask_radius_param_t’; did you mean ‘lv_draw_sw_mask_res_t’?
  128 |     lv_draw_sw_mask_radius_param_t mask_rout_param;
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |     lv_draw_sw_mask_res_t
/home/dha/Software/lvgl/lvgl/src/draw/sw/lv_draw_sw_box_shadow.c:131:9: warning: implicit declaration of function ‘lv_draw_sw_mask_radius_init’; did you mean ‘lv_draw_sw_mask_rect’? [-Wimplicit-function-declaration]
  131 |         lv_draw_sw_mask_radius_init(&mask_rout_param, &bg_area, r_bg, true);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         lv_draw_sw_mask_rect
/home/dha/Software/lvgl/lvgl/src/draw/sw/lv_draw_sw_box_shadow.c:185:42: warning: implicit declaration of function ‘lv_draw_sw_mask_apply’; did you mean ‘lv_draw_sw_mask_rect’? [-Wimplicit-function-declaration]
  185 |                     blend_dsc.mask_res = lv_draw_sw_mask_apply(masks, mask_buf, clip_area_sub.x1, y, w);
      |                                          ^~~~~~~~~~~~~~~~~~~~~
      |                                          lv_draw_sw_mask_rect
/home/dha/Software/lvgl/lvgl/src/draw/sw/lv_draw_sw_box_shadow.c:550:9: warning: implicit declaration of function ‘lv_draw_sw_mask_free_param’; did you mean ‘lv_draw_sw_mask_rect’? [-Wimplicit-function-declaration]
  550 |         lv_draw_sw_mask_free_param(&mask_rout_param);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |         lv_draw_sw_mask_rect
make[2]: *** [lvgl/CMakeFiles/lvgl.dir/build.make:1026: lvgl/CMakeFiles/lvgl.dir/src/draw/sw/lv_draw_sw_box_shadow.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:98: lvgl/CMakeFiles/lvgl.dir/all] Error 2
make: *** [Makefile:136: all] Error 2


### How to reproduce?

lv_conf.h
...
#define LV_DRAW_SW_COMPLEX 0
...

--- a/src/draw/sw/lv_draw_sw_arc.c
+++ b/src/draw/sw/lv_draw_sw_arc.c
@@ -270,5 +270,16 @@ static void get_rounded_area(int16_t angle, int32_t radius, uint8_t thickness, l
     }
 }
 
+#else /*LV_DRAW_SW_COMPLEX*/
+
+void lv_draw_sw_arc(lv_draw_unit_t * draw_unit, const lv_draw_arc_dsc_t * dsc, const lv_area_t * coords)
+{
+    LV_UNUSED(draw_unit);
+    LV_UNUSED(dsc);
+    LV_UNUSED(coords);
+
+    LV_LOG_WARN("LV_DRAW_SW_COMPLEX needs to be enabled");
+}
+
 #endif /*LV_DRAW_SW_COMPLEX*/
 #endif /*LV_USE_DRAW_SW*/

--- a/src/draw/sw/lv_draw_sw_box_shadow.c
+++ b/src/draw/sw/lv_draw_sw_box_shadow.c
@@ -9,6 +9,8 @@
 #include "lv_draw_sw.h"
 #if LV_USE_DRAW_SW
 
+#if LV_DRAW_SW_COMPLEX
+
 #include "blend/lv_draw_sw_blend.h"
 #include "../../core/lv_global.h"
 #include "../../misc/lv_math.h"
@@ -34,11 +36,9 @@
 /**********************
  *  STATIC PROTOTYPES
  **********************/
-#if LV_DRAW_SW_COMPLEX
 static void /* LV_ATTRIBUTE_FAST_MEM */ shadow_draw_corner_buf(const lv_area_t * coords, uint16_t * sh_buf, int32_t s,
                                                                int32_t r);
 static void /* LV_ATTRIBUTE_FAST_MEM */ shadow_blur_corner(int32_t size, int32_t sw, uint16_t * sh_ups_buf);
-#endif /*LV_DRAW_SW_COMPLEX*/
 
 /**********************
  *  STATIC VARIABLES
@@ -552,13 +552,10 @@ void lv_draw_sw_box_shadow(lv_draw_unit_t * draw_unit, const lv_draw_box_shadow_
     lv_free(sh_buf);
     lv_free(mask_buf);
 }
-#endif /*LV_USE_DRAW_SW*/
 
 /**********************
  *   STATIC FUNCTIONS
  **********************/
-
-#if LV_DRAW_SW_COMPLEX
 
 /**
  * Calculate a blurred corner
@@ -726,4 +723,18 @@ static void LV_ATTRIBUTE_FAST_MEM shadow_blur_corner(int32_t size, int32_t sw, u
 
     lv_free(sh_ups_blur_buf);
 }
+
+#else /*LV_DRAW_SW_COMPLEX*/
+
+void lv_draw_sw_box_shadow(lv_draw_unit_t * draw_unit, const lv_draw_box_shadow_dsc_t * dsc, const lv_area_t * coords)
+{
+    LV_UNUSED(draw_unit);
+    LV_UNUSED(dsc);
+    LV_UNUSED(coords);
+
+    LV_LOG_WARN("LV_DRAW_SW_COMPLEX needs to be enabled");
+}
+
 #endif /*LV_DRAW_SW_COMPLEX*/
+
+#endif /*LV_DRAW_USE_SW*/

--- a/src/draw/sw/lv_draw_sw_mask_rect.c
+++ b/src/draw/sw/lv_draw_sw_mask_rect.c
@@ -117,5 +117,16 @@ void lv_draw_sw_mask_rect(lv_draw_unit_t * draw_unit, const lv_draw_mask_rect_ds
  *   STATIC FUNCTIONS
  **********************/
 
+#else /*LV_DRAW_SW_COMPLEX*/
+
+void lv_draw_sw_mask_rect(lv_draw_unit_t * draw_unit, const lv_draw_mask_rect_dsc_t * dsc, const lv_area_t * coords)
+{
+    LV_UNUSED(draw_unit);
+    LV_UNUSED(dsc);
+    LV_UNUSED(coords);
+
+    LV_LOG_WARN("LV_DRAW_SW_COMPLEX needs to be enabled");
+}
+
 #endif /*LV_DRAW_SW_COMPLEX*/
 #endif /*LV_USE_DRAW_SW*/

--- a/tests/src/lv_test_conf.h
+++ b/tests/src/lv_test_conf.h
@@ -100,6 +100,7 @@ typedef void * lv_user_data_t;
 #elif LV_TEST_OPTION == 1
 #define  LV_COLOR_DEPTH     1
 #define  LV_DPI_DEF         30
+#define  LV_DRAW_SW_COMPLEX 0
 #include "lv_test_conf_minimal.h"
 #endif
 

